src/track/track.cpp

```diff
@@ -204,7 +204,24 @@ QString Track::getCanonicalLocation() const {
     // (copy-on write). But operating on a single instance of QFileInfo
     // might not be thread-safe due to internal caching!
     QMutexLocker lock(&m_qMutex);
-    return TrackRef::canonicalLocation(m_fileInfo);
+
+    // Note: We return here the cached value that was calculated just after 
+    // init this Track object. This avoid repeady use of time 
+    // consuming file IO.
+    // We ignore the case when the user changes a symbolic link to 
+    // point a file to an other location, since this is a user action.
+    // We also don't care if a file disappears while Mixxx is running. Opening 
+    // a non-existent file is already handled and doesn't cause any malfunction.
+    QString loc = TrackRef::canonicalLocation(m_fileInfo);
+    if (loc.isEmpty()) {
+        // we see here an empty path because the file did not exist  
+        // when creating the track object.
+        // The user might have restored the track in the meanwhile.
+        // So try again it again. 
+        m_fileInfo.refresh();
+        QString loc = TrackRef::canonicalLocation(m_fileInfo);
```

Comment:
The outer `loc` that actually gets returned is still empty ;)

Reply:
Ping

Reply:
ups.